### PR TITLE
View/Test menu item should not be visible when no view contributions

### PR DESCRIPTION
**Pull request by westbury**
#### What it does

This PR fixes the problem, as reported in https://github.com/eclipse-theia/theia/issues/7081, where there are no views contributed to the 'test' view container.  The 'Test' menu item appears on the 'View' menu when it should not.

Unfortunately this fixes the problem for browser only.  The changes involved to get the plugin contributions before the Electron menu is set would be very substantial.  Perhaps you won't want this PR at all if it does not fix it for both Browser and Electron.  I would understand that.  I have disabled the menu item in our Electron product by hard-calling `unregisterMenuAction`.

#### How to test

Start up Theia both with one or more plugins that contribute a test view, and with no plugins that contribute a test view.  You should see the menu item only in the former case.

I tested on Linux with this plugin:  https://marketplace.visualstudio.com/items?itemName=ms-python.python

#### Review checklist

- [x ] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by akosyakov** [comment]
> The 'Test' menu item appears on the 'View' menu when it should not.

Ah, I thought it appears in the activity bar. In the menu it is not harmful.

**Comment by vince-fugnitto** [comment]
> > The 'Test' menu item appears on the 'View' menu when it should not.
> 
> Ah, I thought it appears in the activity bar. In the menu it is not harmful. It is the same as problems view without any markers, no?

Perhaps we can take it a step further and  also display an alert in the widget itself when no test providers are contributed (similarly to how the `SCM` view displays a **No repository found** message). This way if a user ever explicitly opens the view without any providers they will be alerted with some context.

<div align='center'>

Screen Shot 2020-02-06 at 9 25 46 AM


</div>

File: packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts [hunk 1]
```diff
@@ -615,6 +630,10 @@ export class PluginViewRegistry implements FrontendApplicationContribution {
         }
     }
 
+    private isElectron(): boolean {
```
Comment (marechal-p) [inline packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts]: Either mark this protected or directly use `environment.electron.is()`. It doesn't really add value here.

**Comment by westbury** [comment]
> Perhaps we can take it a step further and also display an alert in the widget itself when no test providers are contributed

Yes, I was thinking along the same lines.  I think I will close this PR for now and consider work to improve the test view as another PR.

**Comment by westbury** [comment]
> Ah, I thought it appears in the activity bar. In the menu it is not harmful. It is the same as problems view without any markers, no?

Yes, it is the same as the problem markers.  The initial feeling here was that new menu items should not unexpectedly appear that do nothing except open an empty dead view.  But I think it is fine to expect extenders to explicitly remove the menu item if they don't want it, so I'm closing this PR.

**Comment by vince-fugnitto** [comment]
> > Perhaps we can take it a step further and also display an alert in the widget itself when no test providers are contributed
> 
> Yes, I was thinking along the same lines. I think I will close this PR for now and consider work to improve the test view as another PR.

I don't have any strong opinions so I'll leave the decision up to you :)

I was thinking along the line that displaying a message would be the simplest solution while also providing feedback to end-users. I wonder if removing the view like this pull-request does would potentially confuse users (for example if they see documentation of the `test view` but actually don't see it in their application).

**Comment by akosyakov** [comment]
> But I think it is fine to expect extenders to explicitly remove the menu item if they don't want it, so I'm closing this PR.

@westbury if something can be done to easy removing it for you in ARM Mbed Studio, sure let's do it

**Comment by westbury** [comment]
> @westbury if something can be done to easy removing it for you in ARM Mbed Studio, sure let's do it

It's easy for us to remove menu items.  We already have a few we remove so it is just one line of code:
`menus.unregisterMenuAction('plugin-view-container.test.toggle');`
So no problem for us.

**Pull request closed without merging**